framework/vulkan_sample.cpp

```diff
@@ -122,7 +122,12 @@ bool VulkanSample::prepare(Platform &platform)
 	}
 #endif
 
-	instance = std::make_unique<Instance>(get_name(), get_instance_extensions(), get_validation_layers(), headless, api_version);
+	create_instance();
+
+	if (!instance)
+	{
+		instance = std::make_unique<Instance>(get_name(), get_instance_extensions(), get_validation_layers(), headless, api_version);
```

Comment:
maybe it would be cleaner to move this to the default implementation of `create_instance`?

Reply:
Not sure if that's possible. The default implementation is just an empty function that can be overridden by a sample.